[WIP] Build with Swift Package Manager

harlanhaskins opened the pull request:
This patch migrates Trill to a modular build system with a CMake driver
that invokes swift build with the appropriate flags to build the LLVM
wrappers. It also includes a build script so contributors don't need to
use CMake.

Closes #2.

harlanhaskins committed:
Build entire project with Swift Package Manager

This patch migrates Trill to a modular build system with a CMake driver
that invokes swift build with the appropriate flags to build the LLVM
wrappers. It also includes a build script so contributors don't need to
use CMake.

harlanhaskins committed:
Actually build in circle

harlanhaskins committed:
Remove CMake and instead perform all logic in utils/build

harlanhaskins committed:
Remove unused variable

harlanhaskins committed:
Replace CMake with build script

harlanhaskins committed:
Fix Circle build

harlanhaskins committed:
Fix Circle build

segiddins commented on Sources/AST/ASTDumper.swift:
```diff
@@ -87,243 +93,264 @@ class ASTDumper<StreamType: ColoredStream>: ASTTransformer {
 }
+
+extension String {
+    func splitCapitals() -> [String] {
```
this is very not-unicode aware. Couldn't you instead split by characters, and check to see if `c.uppercased == c`?

harlanhaskins committed:
Add pkg-config to circle.yml

harlanhaskins committed:
Fix build script string format issue

harlanhaskins committed:
Update circle.yml

harlanhaskins committed:
Update circle.yml

harlanhaskins commented:
Okay so it builds but the build script needs more work to actually report rest failures as failures!

harlanhaskins committed:
Update build script to fail when the first subcommand fails

segiddins commented on Sources/AST/WitnessTable.swift:
```diff
@@ -0,0 +1,20 @@
+import Source
+
+public struct WitnessTable {
+    public let proto: ProtocolDecl
```
indentation

harlanhaskins replied:
Thanks

segiddins commented on README.md:
````diff
@@ -32,42 +27,33 @@ func main() {
-  println(Date().getTime())
-}
+```bash
+swift utils/make-pkgconfig.swift
````
this seems to be gone?

harlanhaskins replied:
I’m debating making that file...
I could just merge the make-pkgconfig files from LLVMSwift and ClangSwift and just put them in this repo?

segiddins commented on README.md:
````diff
@@ -32,42 +27,33 @@ func main() {
-- LLVMSwift and TinyGC, which are vendored as git submodules. Run `git
-  submodule update --init` to clone them.
+```bash
+usage: build-script [-h] [--swift SWIFT] [-r RELEASE] [-x] [-t]
````
I'm not seeing this build script anywhere?

harlanhaskins replied:
I need to rename that to `utils/build`

segiddins reviewed (commented):
```
/Users/segiddins/Development/OpenSource/Trill/Sources/LLVMWrappers/src/LLVMWrappers.cpp:21:10: fatal error: 'clang-c/Platform.h' file not found
#include <clang-c/Platform.h>
         ^~~~~~~~~~~~~~~~~~~~
1 error generated.
```

segiddins committed:
Include runpath search paths when building

segiddins committed:
Print LLVM fatal errors to STDERR

segiddins committed:
Link the runtime when running the JIT

segiddins committed:
Store SourceFile in locations rather than bare file names

segiddins committed:
Handle when importing from files that are not real

segiddins committed:
Remove GC & build the runtime with swiftpm

segiddins committed:
[Lexer] Fix lexing multiline comments that contain / or * before the terminator

segiddins committed:
[build] Ensure the lib dir exists before copying to it

segiddins committed:
[build] Copy files instead of moving them

segiddins committed:
[Package.swift] Re-order deps to patch resolution

segiddins committed:
[README] Mention that trill types will leak

segiddins committed:
Relocate Sources/trillRuntime back to runtime

segiddins committed:
[runtime] Ensure private files can also be included in C

segiddins committed:
[demangle] Update to use string subscripting

segiddins committed:
[build] Build trill-demangle via swiftpm

segiddins committed:
Replace string slicing with subscripting to remove warnings

segiddins committed:
[build] Add machinery for running swift test

segiddins committed:
Merge pull request #63 from trill-lang/seg-remove-gc-build-runtime-spm

Remove GC & build the runtime with swiftpm

segiddins committed:
[Source] Add a SourceFileManager to lazily load file contents (#64)

* [build] use correct LD_RUNPATH_SEARCH_PATHS setting

* [SourceFile] Implement Equatable

* [SourceLocation] Ensure that files are equal

Also assert that SourceRange make sense when built (same file && start <= end)

* [ClangImporter] Cache imported SourceFile structs

This is a roughly 10x performance improvement for the pass

* [SourceFile] Implement Hashable

* [Source] Add a SourceFileManager to lazily load file contents

* [Source] Address PR feedback

* [SourceFileManager] Protect the caches via a dispatch queue

* [ClangInvocation] Mark all linker flags as such

* [build] Ensure generated Xcode project can link libclang

segiddins committed:
[build] Create an Xcodeproj that has all the correct settings / phases (#65)

segiddins committed:
[build] Add a helper for making pckgconfigs

harlanhaskins commented:
Okay here's what I'm gonna do. I'm gonna merge this, then rebase the overloading PR on this. THEN I'm gonna fix the build errors as part of that PR to get all of this unstuck.

segiddins commented:
💕
